test/common/dns.js

```diff
@@ -0,0 +1,290 @@
+'use strict';
+
+// Naïve DNS parser/serializer.
+
+require('../common');
+const assert = require('assert');
+const os = require('os');
+
+const types = {
+  A: 1,
+  AAAA: 28,
+  NS: 2,
+  CNAME: 5,
+  SOA: 6,
+  PTR: 12,
+  MX: 15,
+  TXT: 16,
+  ANY: 255
+};
+
+const classes = {
+  IN: 1
+};
+
+function readDomainFromPacket(buffer, offset) {
+  assert.ok(offset < buffer.length);
+  const length = buffer[offset];
+  if (length === 0) {
+    return { nread: 1, domain: '' };
+  } else if ((length & 0xC0) === 0) {
+    offset += 1;
+    const chunk = buffer.toString('ascii', offset, offset + length);
+    // Read the rest of the domain.
+    const { nread, domain } = readDomainFromPacket(buffer, offset + length);
+    return {
+      nread: 1 + length + nread,
+      domain: domain ? `${chunk}.${domain}` : chunk
+    };
+  } else {
+    // Pointer to another part of the packet.
+    assert.strictEqual(length & 0xC0, 0xC0);
+    // eslint-disable-next-line
```

Comment:
How about:
```js
const kLengthMask = 0xC0;
const kOffsetMask = ~0xC000;
```
I get the idiom, but I have to use my fingers to figure out `0xC000 === 0o‭140000‬ == 0b1100000000000000 === 0x3 << 14` and my mind just has to do that. Using a `const` will allow me to abstract it away 😅